### PR TITLE
Clang compile

### PR DESCRIPTION
**Pull request by dimbleby**
Not entirely sure why clang finds that it needs to know about more code
than gcc - but it does.

### CODE REVIEW

File: src/Makefile [hunk 1]
```diff
@@ -110,7 +112,7 @@ memento_test_LDFLAGS := ${COMMON_LDFLAGS} -lcurl
 
 include ../build-infra/cpp.mk
 
-ROOT := $(abspath $(shell pwd)/../)
+ROOT := ..
```
Comment (bossmc) [inline src/Makefile]: Why this change?  If you want this to affect the `cpp.mk` snippet doesn't it have to be above the `include`?

Also have you checked that the `alarmdefintions.h` file is being correctly generated with this change?

Comment (dimbleby) [inline src/Makefile]: I'm not changing this for `cpp.mk` here, I'm changing it for `alarm-utils.mk` a couple of lines below (but not show by github by default).

Yes, the build is fine with this.

Comment (bossmc) [inline src/Makefile]: Cool, that's fine, I was just checking.
